### DISCUSSION TITLE
Auto expand control type object with providing a boolean "autoexpand: true / false"

### DISCUSSION THREAD

**Discussion by mesihtasci** [storybookjs/storybook#23309] [Ideas]
I would like to have a feature, where an option can be passed to the control panel, where an object type control can be auto expanded recursively, as it is a pain if you have a complex object. You dont want to expand all of the children manually to change something. It would be best to have a parameter called **autoexpand** and we could set it to true or false.

Currently it would look like this:
![current](https://user-images.githubusercontent.com/21260471/126752796-4456fab0-3498-4976-94ab-275296801f23.png)

In future it could look like this, depending on what value the user uses in the **autoexpand** parameter.
![future](https://user-images.githubusercontent.com/21260471/126752794-57f1ca8c-c946-4187-bd3c-d68051fdac8b.png)

**Comment by tripolskypetr** [comment]
+1

**Comment by tripolskypetr** [comment]
I think for now monkey-patch the html with [auto-clicker script](https://github.com/react-declarative/react-declarative-storybook/blob/master/template/object-auto-expand.js) will be the simplest. If you want, you can use [this project as a sample](https://github.com/react-declarative/react-declarative-storybook)

```
const observer = new MutationObserver(() => {
    const expandButtons = [...document.querySelectorAll('button[type=button] > span')]
        .filter((el) => el.innerHTML === 'RAW')
        .map((el) => el.closest('button'));
    for (const button of expandButtons) {
        if (!button.classList.contains(EXPANDED_CLASS)) {
            button.classList.add(EXPANDED_CLASS);
            button.click();
        }
    }
});
observer.observe(document.body, {
    childList: true,
    subtree: true,
});
```

**P.S.** Check webpack config in [`config/main`](https://github.com/react-declarative/react-declarative-storybook/blob/1be4c67c3417606d7d57b6eea3695747e7d8c929/config/main.js). After production build, the patch will be installed automatically

**Comment by shilman** [comment]
Converting this to a discussion. We’re reserving feature issues for things that are on our [short-term roadmap](https://github.com/storybookjs/storybook/milestones). We’ll check back periodically to see whether this is a good candidate to be added. If this feature would be useful to you, please upvote!
